katharsis-core/src/main/java/io/katharsis/core/internal/utils/MethodCache.java

```diff
@@ -0,0 +1,64 @@
+package io.katharsis.core.internal.utils;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.apache.commons.lang3.ObjectUtils;
+
+import io.katharsis.utils.Optional;
+
+public class MethodCache {
+
+	private Map<MethodCacheKey, Optional<Method>> cache = new ConcurrentHashMap<>();
```

Comment:
Despite the benefits of using the guava cache, I'm with remo on this one.  Without a profile marking this cache as an issue, we shouldn't be adding dependencies.  Adding a function to call on a cache miss with Java 8 is no longer as heavy lifting a task as it used to be, and could be added if necessary.